### PR TITLE
chore: update regex for more valid scripture feature references

### PR DESCRIPTION
**Pull request by nlewis84**
## 🐛 Issue

Some scripture features are bombing out in `web-embeds` but not in mobile/admin

## ✏️ Solution

- Adjust the logic of the scripture parsing in `web-embeds`
- Also, it became apparent that there were some styling differences for HtmlFeatures. This was an easy fix, so I just threw it on this PR.

## 🔬 To Test

1. Start the `micro-service` by `yarn dev` in the `/micro-service` folder
2. Nav to http://localhost:3000/?id=UniversalContentItem-8aa3cff4-5054-4464-b1dc-e2e31c5adf1d
3. See 4 example ScriptureFeatures working correctly
4. If you want to verify on the mobile app, use Apollos Demo, head to the Explore tab, check out the last feature and tap on Psalm 100 Devotional

## 📸 Screenshots

| Before |
| --- |
| Screenshot 2024-01-30 at 5 13 45 PM |
| --- |
| After |
| Screenshot 2024-01-30 at 5 49 08 PM |

After photos for mobile and web to show parity
| Mobile | Web |
|---|---|
|  | html styling web |
|  | header styling mobile |

### CODE REVIEW

**Comment by nlewis84** [comment]
@vinnyjth I will work on this description. Discovered this issue while hooking up ScriptureFeatures on Admin. I dropped the before/after photos in the description already. Ideally, I think I would also like to make sure the padding/margin around scripture features on `embeds` matches `mobile`. My gut says it is probably too compact here.

**Comment by vinnyjth** [comment]
@nlewis84 Sounds good! long term, I wonder if we should expose this from the API :D

**Comment by nlewis84** [comment]
> @nlewis84 Sounds good! long term, I wonder if we should expose this from the API :D

Yeah, I think you are right on long term plan for it.

File: packages/web-shared/components/FeatureFeed/Features/ScriptureFeature.js [hunk 1]
```diff
@@ -21,16 +21,37 @@ function ScriptureFeature(props = {}) {
...
+    if (endChapter || endVerse) {
+      // Case for a range
+      if (endChapter && !endVerse) {
+        // Range of chapters, e.g., 'Genesis 1-3'
+        title = `${book} ${startChapter}-${endChapter}`;
+        verses = `Chapters ${startChapter}-${endChapter}`;
+      } else if (startChapter !== endChapter) {
+        // Range of chapters and verses, e.g., 'Genesis 1:1-3:24'
+        title = `${book} ${startChapter}:${startVerse}-${endChapter}:${endVerse}`;
+        verses = `Verses ${startChapter}:${startVerse}-${endChapter}:${endVerse}`;
```
Comment (vinnyjth) [inline packages/web-shared/components/FeatureFeed/Features/ScriptureFeature.js:44]: small nitpik - for clarity that we reuse this text :D 
```suggestion
        verses = `Verses ${title}`;
```

Comment (vinnyjth) [inline packages/web-shared/components/FeatureFeed/Features/ScriptureFeature.js:44]: small nitpik - for clarity that we reuse this text :D 
```suggestion
        verses = `Verses ${title}`;
```

File: packages/web-shared/components/FeatureFeed/Features/ScriptureFeature.js [hunk 2]
```diff
@@ -21,16 +21,37 @@ function ScriptureFeature(props = {}) {
   };
 
   function parseBibleReference(reference) {
-    const regex = /^([\w\s]+)\s(\d+):(\d+)-?(\d+)?$/;
+    // This regex handles cases like 'Genesis 1-3' and 'Genesis 1:1-3:24'
+    const regex = /^([\w\s]+)\s(\d+)(?::(\d+))?(?:-(\d+)(?::(\d+))?)?$/;
```
Comment (vinnyjth) [inline packages/web-shared/components/FeatureFeed/Features/ScriptureFeature.js:25]: 💪

File: packages/web-shared/ui-kit/Longform/Longform.styles.js [hunk 3]
```diff
@@ -7,19 +7,27 @@ import { TypeStyles } from '../Typography';
 const Longform = styled.div`
   ${TypeStyles.BodyText};
   max-width: 700px;
+  margin-bottom: ${themeGet('space.xl')};
```
Comment (vinnyjth) [inline packages/web-shared/ui-kit/Longform/Longform.styles.js:10]: thanks for throwing these style updates in!

**Review by vinnyjth — APPROVED** [review]
Thanks for the helpful examples here!

File: packages/web-shared/ui-kit/Longform/Longform.styles.js (hunk 3 above)
Comment (nlewis84) [reply packages/web-shared/ui-kit/Longform/Longform.styles.js:10]: Yep! Very obvious compared to this before shot:
Screenshot 2024-01-31 at 4 12 11 PM

File: packages/web-shared/components/FeatureFeed/Features/ScriptureFeature.js (hunk 1 above)
Comment (nlewis84) [reply packages/web-shared/components/FeatureFeed/Features/ScriptureFeature.js:44]: Screenshot 2024-01-31 at 4 19 11 PM

Like this?

Comment (vinnyjth) [reply packages/web-shared/components/FeatureFeed/Features/ScriptureFeature.js:44]: 🤦 I missed that the second sample doesn't include the book. What you have is perfect

**Pull request merged**